activerecord/test/cases/associations/nested_through_associations_test.rb

```diff
@@ -639,6 +639,19 @@ def test_polymorphic_has_many_through_joined_different_table_twice
     assert_equal hotel, Hotel.joins(:cake_designers, :drink_designers).take
   end
 
+  def test_has_many_through_polymorphic_with_where_without_table_reference
+    Post.delete_all
+
+    post = Post.create!(title: "Catchy Title", body: "Interesting body.")
+    category = Category.create!(name: "Anything")
+    Post::CategoryPost.create!(post: post, category: category)
+
+    author = authors(:bob)
+    Essay.create!(writer: author, category: category)
+
+    assert_equal 1, Post.joins(:authors_of_essays_named_bob).count
+  end
```

Comment:
This is a perfect replication of the scenario in https://github.com/rails/rails/issues/34613 and results in a failing test. See comment below.